src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveCMarshal.NativeAot.cs

```diff
@@ -11,6 +11,10 @@ namespace System.Runtime.InteropServices.ObjectiveC
     public static unsafe partial class ObjectiveCMarshal
     {
         private static readonly IntPtr[] s_ObjcMessageSendFunctions = new IntPtr[(int)MessageSendFunction.MsgSendSuperStret + 1];
+        private static bool s_initialized;
+        private static readonly ConditionalWeakTable<object, ObjcTrackingInformation> s_objects = new();
+        private static IntPtr s_IsTrackedReferenceCallback;
```

Comment:
Up to you, but you could fully type these callbacks using C# function pointers - `delegate* unmanaged<...>`.

Reply:
Good point, fixed.